js/privatebin.js

```diff
@@ -4149,65 +4485,197 @@ jQuery.PrivateBin = (function($, sjcl, Base64, RawDeflate) {
+        var me = {};
+
         /**
-         * initialize
+         * blacklist of UserAgents (parts) known to belong to a bot
          *
-         * @name   PasteDecrypter.init
+         * @private
+         * @enum   {Array}
+         * @readonly
+         */
+        const badBotUA = [
+            'Bot',
+            'bot'
+        ];
+
+        /**
+         * blacklist of UserAgent versions known not to work with this application
+         *
+         * @private
+         * @enum   {Object}
+         * @readonly
+         */
+        const oldUA = [
+            {
+                'regex': /Chrome\/([0-9]+)/,
+                'minVersion': 57,
+            },
+            {
+                'regex': /Edge\/([0-9]+)/,
+                'minVersion': 16,
+            },
+            {
+                'regex': /Firefox\/([0-9]+)/,
+                'minVersion': 54,
+            },
+            {
+                'regex': /Opera\/.*Version\/([0-9]+)/,
+                'minVersion': 44,
+            },
+            {
+                'regex': /Version\/([0-9]+).*Safari/,
+                'minVersion': 11,
+            }
+        ];
+
+        /**
+         * check if the connection is insecure
+         *
+         * @private
+         * @name   InitialCheck.isInsecureConnection
          * @function
+         * @return {bool}
+         */
+        function isInsecureConnection()
+        {
+            const url = new URL(document.URL);
+
+            // HTTP is obviously insecure
+            if (url.protocol !== 'http:') {
+                return false;
+            }
+
+            // filter out actually secure connections over HTTP
+            if (
+                url.hostname.endsWith('.onion') ||
+                url.hostname.endsWith('.i2p')
+            ) {
+                return false;
+            }
+
+            // whitelist localhost for development
+            if (
+                url.hostname === 'localhost' ||
+                url.hostname === '127.0.0.1'
+            ) {
+                return false;
+            }
+
+            // totally INSECURE http protocol!
+            return true;
+        }
+
+        /**
+         * checks whether this is a bot we dislike
+         *
+         * @private
+         * @name   InitialCheck.isBadBot
+         * @function
+         * @return {bool}
+         */
+        function isBadBot() {
+            // check whether a bot user agent part can be found in the current
+            // user agent
+            for (let i = 0; i < badBotUA.length; ++i) {
+                if (navigator.userAgent.indexOf(badBotUA) >= 0) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        /**
+         * checks whether this is an unsupported browser
+         *
+         * @private
+         * @name   InitialCheck.isOldBrowser
+         * @function
+         * @return {bool}
+         */
+        function isOldBrowser() {
```

Comment:
ah, oaky, this is a new feature(?)… at least I did not added it in this PR AFAIK.

So maybe adjust PR title.

Reply:
Yes and no: This [used to be targetted just at IE < 10 via a browser switch in HTML](https://github.com/PrivateBin/PrivateBin/commit/fc914b4b8471287e6f7fb9c6511c8a2036801711#diff-462550ddcf082451651ab9c6f16a9702L76). I added the version list [based on my research in the webcrypto PR](https://github.com/PrivateBin/PrivateBin/pull/431#issuecomment-493129359).